src/modules/core/components/NetworkStatus/NetworkStatus.tsx

```diff
@@ -0,0 +1,169 @@
+import cx from 'classnames';
+import React, { useState, useEffect } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import styles from './NetworkStatus.css';
+import {
+  useLatestRpcBlockQuery,
+  useColonyServerLivenessQuery,
+  useLatestSubgraphBlockQuery,
+  useReputationOracleLivenessQuery,
+} from '~data/index';
+
+import Popover from '~core/Popover';
+import Icon from '~core/Icon';
+import ExternalLink from '~core/ExternalLink';
+import { NETWORK_HEALTH } from '~externalUrls';
+
+const MSG = defineMessages({
+  networkStatusButton: {
+    id: 'pages.NavigationWrapper.UserNavigation.networkStatusButton',
+    defaultMessage: `Network`,
+  },
+
+  networkHealthHeader: {
+    id: 'pages.NavigationWrapper.UserNavigation.networkHealthHeader',
+    defaultMessage: `Network health is {networkHealth}`,
+  },
+
+  networkHealthDescription: {
+    id: 'pages.NavigationWrapper.UserNavigation.networkHealthDescription',
+    defaultMessage: `{networkHealth, select,
+      poor
+        {You should be able to perform actions, however,
+        there is problem retrieving new information from the
+        chain. We are working to resolve this.}
+      critical
+        {You will have trouble performing actions and retrieving
+        information from the chain. We are working to resolve this.}
+        other {Network is healthy.}}`,
+  },
+});
+
+const NetworkStatus = () => {
+  enum NetworkHealthType {
+    Healthy = 'healthy',
```

Comment:
I think what Olga meant here is to move them outside of the component **file** as well, and just import them, usually inside `src/types`

That way they can be share by the whole dapp, in case there's a need in the future to touch, or integrate, parts of the server health system somewhere else